cli/generate/src/render.rs

```diff
@@ -115,7 +116,7 @@ impl Generator {
         self.add_non_terminal_alias_map();
         self.add_primary_state_id_list();
 
-        if !self.supertype_map.is_empty() {
+        if !self.supertype_map.is_empty() && self.abi_version >= ABI_VERSION_WITH_SUPERTYPE_INFO {
```

Comment:
Can the entire `supertype_map` initialisation be gated behind the ABI version check?
And shouldn't the C API also have a similar check?